postgres-protocol/src/types/numeric.rs

```diff
@@ -0,0 +1,1583 @@
+//! Conversions to and from Postgres's binary format for the numeric type.
+use byteorder::{BigEndian, ReadBytesExt};
+use bytes::{BufMut, BytesMut};
+use std::boxed::Box as StdBox;
+use std::error::Error;
+use std::str::{self, FromStr};
+
+/// Serializes a `NUMERIC` value.
+#[inline]
+pub fn numeric_to_sql(v: Numeric, buf: &mut BytesMut) {
+    let num_digits = v.digits.len() as u16;
+    buf.put_u16(num_digits);
+    buf.put_i16(v.weight);
+    buf.put_u16(v.sign.into_u16());
+    buf.put_u16(v.scale);
+
+    for digit in v.digits {
+        buf.put_i16(digit);
+    }
+}
+
+/// Deserializes a `NUMERIC` value.
+#[inline]
+pub fn numeric_from_sql(mut buf: &[u8]) -> Result<Numeric, StdBox<dyn Error + Sync + Send>> {
+    let num_digits = buf.read_u16::<BigEndian>()?;
+    let mut digits = Vec::with_capacity(num_digits.into());
+
+    let weight = buf.read_i16::<BigEndian>()?;
+    let sign = NumericSign::try_from_u16(buf.read_u16::<BigEndian>()?)?;
+
+    let scale = buf.read_u16::<BigEndian>()?;
+
+    for _ in 0..num_digits {
+        digits.push(buf.read_i16::<BigEndian>()?);
+    }
+
+    Ok(Numeric {
+        sign,
+        scale,
+        weight,
+        digits,
+    })
+}
+
+/// Numeric sign
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum NumericSign {
+    /// Positive number
+    Positive,
+    /// Negative number
+    Negative,
+    /// Not a number
+    NaN,
+    /// Positive infinity
+    PositiveInfinity,
+    /// Negative infinity
+    NegativeInfinity,
+}
+
+impl NumericSign {
+    #[inline]
+    fn try_from_u16(sign: u16) -> Result<NumericSign, StdBox<dyn Error + Sync + Send>> {
+        match sign {
+            0x0000 => Ok(NumericSign::Positive),
+            0x4000 => Ok(NumericSign::Negative),
+            0xC000 => Ok(NumericSign::NaN),
+            0xD000 => Ok(NumericSign::PositiveInfinity),
+            0xF000 => Ok(NumericSign::NegativeInfinity),
+            _ => Err("invalid sign in numeric value".into()),
+        }
+    }
+
+    #[inline]
+    fn into_u16(self) -> u16 {
+        match self {
+            NumericSign::Positive => 0x0000,
+            NumericSign::Negative => 0x4000,
+            NumericSign::NaN => 0xC000,
+            NumericSign::PositiveInfinity => 0xD000,
+            NumericSign::NegativeInfinity => 0xF000,
+        }
+    }
+}
+
+/// A Posgres numeric
```

Comment:
Copies the description from `PgLsn`.

```suggestion
/// Postgres `NUMERIC` type.
```